dot/state/block_finalisation.go

```diff
@@ -235,8 +235,17 @@ func (bs *BlockState) handleFinalisedBlock(curr common.Hash) error {
 			return err
 		}
 
-		// the block will be deleted from the unfinalisedBlockMap in the pruning loop
-		// in `SetFinalisedHash()`, which calls this function
+		// delete from the unfinalisedBlockMap and delete reference to in-memory trie
+		block, has = bs.getAndDeleteUnfinalisedBlock(hash)
+		if !has {
+			continue
+		}
+
+		logger.Tracef("cleaned out finalised block from memory; block number %s with hash %s", block.Header.Number, hash)
+
+		go func(header *types.Header) {
+			bs.pruneKeyCh <- header
+		}(&block.Header)
```

Comment:
Do we really need to run this in a goroutine? Can this be blocking :thinking:

Reply:
no it shouldn't be blocking anymore, will update

Reply:
woops, i updated the other part but not here. done!